channeldb/invoices.go

```diff
@@ -2761,6 +2761,102 @@ func delAMPSettleIndex(invoiceNum []byte, invoices,
 	return nil
 }
 
+// DeleteCanceledInvoices deletes all canceled invoices from the database.
+func (d *DB) DeleteCanceledInvoices(_ context.Context) error {
+	return kvdb.Update(d, func(tx kvdb.RwTx) error {
+		invoices := tx.ReadWriteBucket(invoiceBucket)
+		if invoices == nil {
+			return nil
+		}
+
+		invoiceIndex := invoices.NestedReadWriteBucket(
+			invoiceIndexBucket,
+		)
+		if invoiceIndex == nil {
+			return invpkg.ErrNoInvoicesCreated
```

Comment:
Think we should unify the behaviors here - in `FetchPendingInvoices` we'd ignore this error, and we should. do the same here? In general I think `DeleteCanceledInvoices` should be a NOOP if there are no invoices in the db, instead of returning an error.

Reply:
This case is different though right? We do return `nil` above when we "open" the read-write bucket `invoices` however if that exist, we should assume that both the the invoice index and add index exists.

Reply:
could it be the case that this is just an empty bucket? like all the invoices are deleted?

Reply:
I think it's there to give us some reassurance that about expected existence of those buckets. Similar checks are done for other channeldb methods too.

Reply:
cool then I interpret it as a db corruption check.